### DISCUSSION TITLE
Permutation scheme for nested data

### DISCUSSION THREAD

**Discussion by PeBarcik** [gavinsimpson/permute#39] [Q&A] [answered]
Hello all!

I'm having trouble deciding what is the best way to analyse my data.

Here is the description of my study:
I have **6 locations** (blocks) over the country. **In each location, I have 3 plots**. Each plot has one treatment, so basically in each location I have the 3 treatments that I'm studying. I**nside each plot, I took 4 samples** (to have a greater sampling size of my treatments, although I know they are non-independent, and thus require a permutation scheme for a correct analysis).
My objective is to compare treatments, so I don't want to compare samples inside a plot. **I want to compare plots**, and take into considerantion that they are inside blocks (Locations).

For that, I came with this permutation scheme:

```r
perm <- with(data_Q, how(nperm = 999,
                         plots = Plots(strata = Plot, type = "free"),
                         within = Within(type = "none"),
                         blocks = Location))
```

Does this satisfy my objectives?

Also, and I apologize if this is not in the correct place, if I were to use a Linear Mixed Model with this nested data, how would I have to do it to satisfy my objectives?

I came up with this: 

```r
lmer(Y ~ X + (1 | Location) + (0 + X | Location) + (1 | Location:Plot))
```

Should I exclude the `(1|Location:Plot)` ? 
Should I include `(0 + X|Location:Plot)` ?

Thank you!

**Comment by gavinsimpson** [comment, marked as answer]
Yes, your permutation design is an appropriate way to permute data in a design-based framework. You don't say what model you are fitting so I can't comment on whether the deisgn is appropriate given the model.

As for your `lmer()` questions, I don't see why (you haven't explained and this Q&A is *not* the right place for a LMM discussion) you would want uncorrelated random slopes and intercepts. Also, don't you want `(1 | Location / Plot)`, as `Plot` is nested in `Location`, though perhaps `A / B` evaluates to `A:B`, I'd need to check.

**Reply by PeBarcik** [reply]
Thank you Mr. Gavin for your edits (I'm new here and didn't know the formal conventions) and for the answer. 

If I understood well, the `within = Within(type = "none")` is the line of code that says to not compare samples inside the plots, only between plots, am I right?

Also, sorry for not providing the research details earlier, I wanted to spare the readers of the little details, but since it may be important for the discussion, here it is:
I am testing the effect of mixed forests on soil fungal composition and diversity, in comparison to pure forests. Since the nature and design of this is study is very conducive for pseudo-replication mistakes, I needed to "correct" some statistical analyses with a proper permutation design.
The statistical analysis I'm using are PERMANOVA and ANOSIM to compare communities. Also, ANOVA (on a LMM instead of a LM) to compare differences in the diversity.

**Reply by gavinsimpson** [reply]
> If I understood well, the within = Within(type = "none") is the line of code that says to not compare samples inside the plots, only between plots, am I right?

Yes; you don't want the variation between the replicates to be included in the test.

> The statistical analysis I'm using are PERMANOVA and ANOSIM to compare communities.

Don't use `anosim()`. Instead of PERMANOVA (`adonis2()`) I would use `dbrda()` if you _must_ use a dissimilarity. I would prefer RDA or CCA personally.

> Also, ANOVA (on a LMM instead of a LM) to compare differences in the diversity.

As "diversity" *cannot* be negative, a LMM would be inappropriate. I would use a GLMM via `glmer()` or `glmmTMB::glmmTMB()`

**Reply by PeBarcik** [reply]
> Don't use anosim(). Instead of PERMANOVA (adonis2()) I would use dbrda() if you must use a dissimilarity. I would prefer RDA or CCA personally.

May I ask why?

> As "diversity" cannot be negative, a LMM would be inappropriate. I would use a GLMM via glmer() or glmmTMB::glmmTMB()

I will investigate. Thank you!

**Reply by gavinsimpson** [reply]
> May I ask why?

Why what? I made three statements; do you want an explanation for all three?

**Reply by PeBarcik** [reply]
Yes, I would appreciate. Or at least a link to a previous discussion regarding those topics. I'm quite new to statistics, and don't really understand why some approaches may be better than others.

**Reply by gavinsimpson** [reply]
On `anosim()`, see the *Note* section in `?anosim` and the cited reference.

If you convert to dissimilarities you lose the species (response) information. We can add these "species scores" back on to the ordination but they are not the same as the species scores from RDA or CCA, which do not start from a dissimilarity matrix. Also, you have to be very careful in your choice of dissimilarity; too many use the default, Bray Curtis, which is arguably a bad choice as it conflates size (abundance) and shape (compositional) aspects of the response.

I prefer `dbrda()` over `adonis2()` as the former allows `Condition()` in the formula so we can exclude variation do to nuisance variables or experimental design features. This `Condition()` is not available in `adonis2()`.

**Reply by jarioksa** [reply]
@gavinsimpson do not spread this rubbish: Bray-Curtis is usually a good choice. It does suffer from the same problem as RDA (and all Euclidean methods) if sample totals vary a lot _and_ this variation has no ecological meaning.  For instance in `metaMDS` we therefore use Wisconsin double standardization that involves equalizing totals. Also, if you think that all analyses should be purely compositional, then it follows that CA/CCA should be the only method to use (which is Greenacre's argument), but seemingly you do not think so.

**Reply by gavinsimpson** [reply]
@jarioksa You are reading a lot into what I said. If you want an abundance-based analysis, Bray Curtis can include some composition. If you want a compositional analysis, it can bring in some abundance. More than if we used Euclidean / Manhattan or Chi-square as the dissimilarity directly. This is on raw data.

Transformations change all of this; we had Hellinger as a *transformation* well before it was available as an actual dissimilarity in `vegdist()`, and that is a Euclidean distance on transformed distance, but as the data are normalized by the row totals it is compositional even though we use it with RDA. The problem with BC is that I don't know what it is doing in general (i.e. you'd need to verify how much of the $d_{ij}$ is compositional and how much is abundance (size) based for each data set, and no one does that). Of course, if you transform the data beforehand you will force say a compositional view - BC on Wisconsin double standardized data is quite different from BC on raw data, which is what most people blindly do. We don't offer to do `metaMDS`-like transforms in `adonis2()` unlike in `dbrda()` and `capscale()`, so most naive users *won't* be doing what is arguably better practice. Perhaps we should add `metaMDSdist` argument to `adonis2()`.

I don't think we should do every analysis with CA/CCA exactly because we can be interested in things other than compositional differences. A lot of the examples I use for teaching use RDA and log-transformed (`log1p()`) responses, so I'm certainly not advocating for a CA/CCA-only world.

The problem I see with the approach from many users that we see asking questions about vegan (here, on the vegan GitHub, or SO and CV) is that they just default to PERMANOVA and Bray-Curtis because that's what they see others doing and/or BC is the default.

You will note I suggested they use `dbrda()` instead of `adonis2()` if they want to use a dissimilarity. I did point out a slight issue with dissimilarity-based methods in that the species scores are added as a secondary step (i.e. in a supplementary vein, as we do in PCA/CA/PCO/NMDS + `envfit()` or Canoco's supplementary variables). Whether that's a problem for the OP is something only they can answer; I don't know what their research question is. But it is a valid point to be aware of.

As the OP was asking about constrained methods I didn't mention `metaMDS()`.

**Reply by PeBarcik** [reply]
Thank you so much @gavinsimpson and @jarioksa for your answers and the discussion provided. Unfortunately, I'm quite new to this subject, and cannot answer if I _must_ use dissimilarity, and if adding the species scores as a secondary step is a problem or not. But I'll be aware of your recomendations, and research further into those. 
Once again, I appreciate your time and effort put in this Q&A.

**Reply by gavinsimpson** [reply]
@PeBarcik If you are going to be doing this analysis then it is incumbent upon you to figure out these things before you do the analysis, whether you are new to these kinds of methods or not.

**Reply by PeBarcik** [reply]
Yes, you are right. If I were to have more questions regarding these topics, should I open another discussion on the vegan repository or keep sending in this Q&A?

**Reply by gavinsimpson** [reply]
> If I were to have more questions regarding these topics, should I open another discussion on the vegan repository or keep sending in this Q&A? 

If the question is about vegan specifically then do ask there in the discussions as a Q&A. Likewise if it is about permute specifically, ask here. But if it is a more general statistical question, ask on CrossValidated.
